packages/carbon-web-components/src/components/ui-shell/side-nav.ts

```diff
@@ -71,17 +71,6 @@ class CDSSideNav extends HostListenerMixin(LitElement) {
   // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
   private _handleButtonToggle = async (event: CustomEvent) => {
     this.expanded = event.detail.active;
-    if (this.expanded) {
-      await this._updateAndTransitionPromise;
-      // Checks if the side nav is not collapsed during the animation
-      if (this.expanded) {
-        (
-          this.querySelector(
-            (this.constructor as typeof CDSSideNav).selectorNavItems
-          ) as HTMLElement
-        )?.focus();
-      }
-    }
```

Comment:
While we change the value of `this.expanded` here, we then _immediately_ wait for the resolution of `this.update` from the component's update/render cycle.

`this.update` is already resolved from the last run of the component lifecycle and our current function hasn't been off the main thread to let the component start a new update.

Since we're awaiting a resolved promise, we immediate move on to try to focus on elements that are not yet in the DOM.